File: 1-adhock/3-x-o/solution.go
package main

import "fmt"

func xo(str string) bool {
	counter := 0
	for _, s := range str {
		if string(s) == "x" {
			counter++
		} else if string(s) == "o" {
			counter--
		}
	}
	if counter == 0 {
		return true
	}
	return false
}

func main() {
	fmt.Println(xo("xoxoxo"))   // True
	fmt.Println(xo("oxooxo"))   // False
	fmt.Println(xo("oxo"))      // False
	fmt.Println(xo("xxxooo"))   // True
	fmt.Println(xo("xoxooxxo")) // True
}

File: 1-adhock/3-x-o/solution.js
function xo(str) {
  let counter = 0
  for (s of str) {
    if (s == 'x') {
      counter += 1
    } else if (s == 'o') {
      counter -= 1
    }
  }
  return counter == 0 ? true : false
}

console.log(xo('xoxoxo')) // True
console.log(xo('oxooxo')) // False
console.log(xo('oxo')) // False
console.log(xo('xxxooo')) // True
console.log(xo('xoxooxxo')) // True